src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1,394 +1,625 @@
-                    double cpuThrottleAvg = cpuThrottleResults.map(m -> m.getAggregationInfoResult().getAvg()).orElse(0.0);
-                    double cpuThrottleMax = cpuThrottleResults.map(m -> m.getAggregationInfoResult().getMax()).orElse(0.0);
-                    double cpuThrottleSum = cpuThrottleResults.map(m -> m.getAggregationInfoResult().getSum()).orElse(0.0);
-                    double cpuRequestInterval = 0.0;
-                    double cpuUsagePod = 0;
-                    int numPods = 0;
-
-                    // Use the Max value when available, if not use the Avg
-                    double cpuUsage = (cpuUsageMax > 0) ? cpuUsageMax : cpuUsageAvg;
-                    double cpuThrottle = (cpuThrottleMax > 0) ? cpuThrottleMax : cpuThrottleAvg;
-                    double cpuUsageTotal = cpuUsage + cpuThrottle;
-
-                    // Usage is less than 1 core, set it to the observed value.
-                    if (CPU_ONE_CORE > cpuUsageTotal) {
-                        cpuRequestInterval = cpuUsageTotal;
+        interval_end_time = Utils.DateUtils.getTimeStampFrom(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT,
+                intervalEndTimeStr);
+        LOGGER.debug("UpdateRecommendations API request count: {} experiment_name : {} and interval_start_time : {} and interval_end_time : {} ", calCount, experiment_name, intervalStartTimeStr, intervalEndTimeStr);
+        LOGGER.debug("experiment_name : {} and interval_start_time : {} and interval_end_time : {} ", experiment_name, intervalStartTimeStr, intervalEndTimeStr);
+
+        Map<String, KruizeObject> mainKruizeExperimentMAP = new ConcurrentHashMap<>();
+        Map<String, Terms> terms = new HashMap<>();
+        try {
+            new ExperimentDBService().loadExperimentFromDBByName(mainKruizeExperimentMAP, experiment_name);
+            if (null != mainKruizeExperimentMAP.get(experiment_name)) {
+                kruizeObject = mainKruizeExperimentMAP.get(experiment_name);
+            }
+            if (null != kruizeObject) {
+                // set the default terms if the terms object is empty
+                if (kruizeObject.getTerms() == null)
+                    setDefaultTerms(terms);
+                // get the datasource
+                String dataSource = kruizeObject.getDataSource(); // TODO: This will be required for future use
+                int maxDay = Terms.getMaxDays(terms);
+                interval_start_time = Timestamp.valueOf(Objects.requireNonNull(interval_end_time).toLocalDateTime().minusDays(maxDay));
+
+                mainKruizeExperimentMAP.put(experiment_name, kruizeObject);
+                new ExperimentDBService().loadResultsFromDBByName(mainKruizeExperimentMAP, experiment_name,
+                        interval_start_time, interval_end_time);
+
+                // generate recommendation
+                try {
+                    generateRecommendations(kruizeObject);
```

Comment:
Have abstraction for
Terms and Models
it should take two parameter container Name as a parameter and  <ts,IntervalResults> results
and return recommendations 
should be set Container